client/src/main.rs

```diff
@@ -1687,19 +1686,7 @@ fn listen(matches: &ArgMatches<'_>) {
 								dispatch_error: _,
 								dispatch_info: _,
 							} => {
-								let event_records = vec![evr];
-
-								let decoded_event = Events::new(
-									api.metadata().clone(),
-									Hash::default(),
-									event_records.encode(),
-								)
-								.iter()
-								.next()
-								.unwrap()
-								.unwrap();
-
-								error!("ExtrinsicFailed: {:?}", decoded_event);
+								error!("ExtrinsicFailed: {ee:?}");
```

Comment:
The api-client could serve an api that knows the extrinsic index in a block and by knowing this index, it could really just filter the event associated with that extrinsic. Does it already have that?

Reply:
You mean something like this? https://github.com/scs/substrate-api-client/blob/master/src/api/rpc_api/events.rs#LL67C2-L76C3 resp. https://github.com/scs/substrate-api-client/blob/master/src/api/rpc_api/events.rs#L199-L220

Reply:
Follow up question: What would you like to do with it? Simply print all events in a block? Or just check if the xt has been successful? If the latter,  https://github.com/scs/substrate-api-client/blob/master/src/api/rpc_api/author.rs#L276 is probably better suited.

Reply:
hahah, exactly :D

Reply:
But as this is just in the generic `listen` command, which prints all the events happening onchain, there is not a real improvement that we can do here.